### ISSUE TITLE
Provide an option for making the toolbar location fixed

### ISSUE THREAD

**Issue by joepie91** [flameshot-org/flameshot#2245]
### Feature Description

Note: this is *technically* a partial duplicate of #573 - but I am deliberately filing this as a separate issue, to propose a concrete way to improve it that will likely be low-effort to implement. The discussion over in #573 seems to lack focus, and I'm concerned that the issue's scope is preventing an easy win here.

Currently, the toolbar icons are 'wrapped around' the selected region. Unfortunately, this means that icons are not in a consistent location, and especially with small selections it can be very difficult to identify the correct toolbar icon among the cramped pile of them.

Judging from other issues, there are also people who *do* appreciate the buttons being right near the selection. Therefore, instead of *changing* the toolbar location (and potentially ending up in bikeshedding territory), I'd like to propose making its position *configurable* with a binary switch. Either:

1. Near selection (current behaviour)
2. Fixed to the top of the screen

This would provide a quick usability win for those who care more about muscle memory than about proximity, while having a single obvious implementation (line of fixed-position icons at the top of the screen), sidestepping discussions like "*how* do we best wrap the icons around the region?" for now, so that such improvements can be iterated upon later.

If there's a concern about the toolbar overlapping with the selection, that can be addressed fairly easily by having a 'collapse' button toggle that reduces the toolbar to a very narrow strip until re-expanded. Alternatively, an initial implementation could simply come with the caveat that "if you're concerned about overlap, you should be using the near-selection setting", leaving this to be addressed later.

A simplified mockup of what I'm proposing:

![image](https://user-images.githubusercontent.com/1663259/149348490-95e7a12e-6cbc-4434-9204-5e45e7472de7.png)

**Comment by mmahmoudian** [comment]
> I'd like to propose making its position configurable with a binary switch

I agree with this point. I believe the current behavior of Flameshot is super intuitive for our general audience and new users and it should be kept as such as the default, but adding an option to toggle the UI to a toolbar mode would be a welcome change. I'm not 100% sold on the collapsible top-left bar since multi-monitor users will start to complain, but at the moment I cannot think of a more clean way and easy-to -implement way.

A MacOS floating dock style also comes to mind, with some auto-hide with the feature that it gets visible when mouse touches any of the edges of the screen **on that edge**. The reason I suggest "every edge" is that to make the dock appear on the monitor that the mouse is and also the user has the freedom to choose the closes edge and move the mouse there.

![image](https://user-images.githubusercontent.com/390889/149358302-fb46d92c-225f-4905-92c9-e2dea22fa2ad.png)

In the image above, monitors are illustrated by rectangles with red border
the captured screen is illustrated by a checkerboard pattern
the dock is illustrated near the edges, given that the mouse pointer touches that edge

This way nothing nothing is covering the elements on the page and the tools are accessible from the closest edge.

**Comment by nneul** [comment]
Related to my #2497 (closed as duplicate) one related option that I'd like to see is ability to reorder the buttons, as that would make some of the positioning concerns less critical. (i.e. for me 99% of the time I snap + copy to clip), having that reliably be in the first toolbar button would be ideal.

**Comment by dadaphl** [comment]
I know this issue is stale. But i just wanted to say that the unpredictable location of the icons in small capture regions creates so much friction in my head. I need at least 5 seconds to parse where things are. I love this tool, but I caught myself not wanting to take a screenshot because it takes so much effort to decipher the interface.
